### DISCUSSION TITLE
Blade: how to get path for image asset?

### DISCUSSION THREAD

**Discussion by spaceageliving** [statamic/cms#6268] [General]
What is the correct way to retrieve the path for an image asset in a blade template? The image asset is injected into the view as I can dump it and see it in debugbar, but I cannot for the life of me get the path of the image in my blade template.

**Comment by joshuablum** [comment]
Hey @spaceageliving,

Use the `$page` variable that gets passed into the view.

So it should look something like this:
```blade
{{ $page->hero_image->url }}
```

More on this [in the docs](https://statamic.dev/blade#view-data).

Hope that helps :)

**Reply by spaceageliving** [reply]
Hi @joshuablum,

Thanks for this info....when I try this {{ $page->main_image->url }}, I get:

<img width="906" alt="Screen Shot 2022-06-30 at 10 14 20 AM" src="https://user-images.githubusercontent.com/874600/176727019-6ea568f1-db33-4bea-b5cb-32f3f4df2a29.png">

The jpeg is definitely uploaded and available on disk, and a collection instance is injected on $page->main_image.

**Reply by joshuablum** [reply]
So you can see the image in the asset browser and can open it if you open it directly? What does your Blade template look like?
![CleanShot 2022-06-30 at 6 42 03](https://user-images.githubusercontent.com/3824203/176732538-ed915e4e-0f5e-461c-ae9a-82a6ead55271.png)

**Reply by spaceageliving** [reply]
Yes it is visible in cp/assets and can hit the view button:

![Screen-Shot-2022-06-30-at-11 01 11-AM](https://user-images.githubusercontent.com/874600/176735921-14e26992-25df-4c51-b7db-0fe07c78cf60.jpg)

<img width="1840" alt="Screen Shot 2022-06-30 at 11 00 51 AM" src="https://user-images.githubusercontent.com/874600/176735952-3f302c81-0294-415c-b838-179293344dc6.png">

<img width="1840" alt="Screen Shot 2022-06-30 at 11 00 46 AM" src="https://user-images.githubusercontent.com/874600/176736003-2f6ada83-5686-4d7b-9e7d-9a288806d946.png">

Blade currently looks like this, obviously hardwired at the moment to get it working in my layout:

```
<img class="object-scale-down h-64 mx-auto border border-black" src="/assets/tull-thick-as-a-brick.jpeg" alt="Thick as a Brick">
```

If I change to:

```
<img class="object-scale-down h-64 mx-auto border border-black" src="{{ $page->main_image->url }}" alt="Thick as a Brick">
```

I get the error from above (property does not exist).

**Reply by joshuablum** [reply]
Ah, looks to me like you have multiple assets enabled on the Main Image.  Can you change it to `max_files: 1 ` on the Blueprint and try again? Either manually in the yaml or via the CP.

**Reply by spaceageliving** [reply]
OK I changed the yaml on the blueprint to:

<img width="392" alt="Screen Shot 2022-06-30 at 12 10 32 PM" src="https://user-images.githubusercontent.com/874600/176748066-19f88d22-fa9f-46f2-b43f-086c62dcadd4.png">

Deleted the image from the entry, reselected the image using Browse button.

Still getting the property not present.

Oddly, the control panel won't save the change to max_files to the yaml....and after I added manually to the yaml file, the control panel still doesn't show the value. Not sure what is up there...

I also ran php please stache:clear...no help.

**Reply by jasonvarga** [reply]
What if you change the array in the data to a string?

eg.

```yaml
main_image:
  - image.jpg
```
to

```yaml
main_image: image.jpg
```

**Reply by spaceageliving** [reply]
OK awesome this works! Thanks again @jasonvarga. Just a note for future travelers, I initially put max_items: 1, but for image asset, the yaml needs to be:

```
max_files: 1
```

**Reply by jasonvarga** [reply]
Ah yes `max_files` for assets fields. I think that should have been enough without you needing to change the data like my last comment.

**Reply by spaceageliving** [reply]
You had it correct the first time as max_files, it was my mistake.
